Error reusing a variable set by find_path function in FindConnextDDS.cmake

### Information

- **RTI Product**: ConnextDDS Pro
- **Version**: 6.1.0
- **Operating system**: GNU/Linux
- **Compiler**: gcc
- **Compiler version**: 7.3.0
- **Additional information**: 


### What is the current behavior?

Error when trying to build any example with 6.1.0 version of ConnextDDS

### Steps to reproduce the issue

- set necessary environment variables pointing to the 6.1.0 installation directory
- run the linux_static_analysis.py script in the resources/ci_cd directory
- get the error

```
...
CMake Error in builtin_topics/c/CMakeLists.txt:
Imported target "RTIConnextDDS::c_api" includes non-existent path
"/home/mnunez/rti_connext_dds-6.1.0/include/ndds/hpp/hpp"
in its INTERFACE_INCLUDE_DIRECTORIES.  Possible reasons include:
* The path was deleted, renamed, or moved to another location.
* An install or uninstall procedure did not complete successfully.
* The installation package was faulty and references files it does not
provide.
...
```

### What is the expected behavior?

Compile without errors

### Suggested solutions

Just needs to be changed the variable used to find the base include directories in the FindConnextDDS script

--- a/resources/cmake/FindRTIConnextDDS.cmake
+++ b/resources/cmake/FindRTIConnextDDS.cmake
@@ -957,9 +957,11 @@ function(create_connext_imported_target)
 
     if(WIN32
         AND NOT BUILD_SHARED_LIBS
-        AND ${_CONNEXT_TARGET} STREQUAL "routing_service_c")
+        AND ${_CONNEXT_TARGET} STREQUAL "routing_service_c"
+        AND RTICONNEXTDDS_VERSION VERSION_LESS "6.0.0"
+    )
         # ROUTING-276: Routing Service is not available as a static library
-        # for Windows
+        # for Windows until 6.0.0
         return()
     endif()
 
@@ -1293,7 +1295,9 @@ if(distributed_logger IN_LIST RTIConnextDDS_FIND_COMPONENTS
         set(RTIConnextDDS_distributed_logger_FOUND TRUE)
 
         if(CMAKE_SYSTEM_NAME MATCHES "Linux" AND
-            CMAKE_COMPILER_VERSION VERSION_GREATER "4.6.0")
+            CMAKE_COMPILER_VERSION VERSION_GREATER "4.6.0" AND
+            RTICONNEXTDDS_VERSION VERSION_LESS "6.1.0"
+        )
            set(CONNEXTDDS_EXTERNAL_LIBS
                -Wl,--no-as-needed  ${CONNEXTDDS_EXTERNAL_LIBS})
         endif()
@@ -1361,7 +1365,7 @@ if(routing_service IN_LIST RTIConnextDDS_FIND_COMPONENTS
         "routing_service_host"
         "routing_service"
     )
-    if(RTIConnextDDS_FIND_VERSION VERSION_GREATER 5.3.0.8)
+    if(RTICONNEXTDDS_VERSION VERSION_GREATER 5.3.0.8)
         list(APPEND rti_versions_field_names_host
             "routing_service_sdk_jars"
         )
@@ -1411,9 +1415,10 @@ if(routing_service IN_LIST RTIConnextDDS_FIND_COMPONENTS
     get_all_library_variables("${routing_service_libs}" "ROUTING_SERVICE_API")
 
     if(WIN32 AND ROUTING_SERVICE_API_RELEASE_STATIC AND
-            ROUTING_SERVICE_API_DEBUG_STATIC)
+        ROUTING_SERVICE_API_DEBUG_STATIC AND
+        RTICONNEXTDDS_VERSION VERSION_LESS "6.0.0")
         # ROUTING-276: Routing Service is not available as a static library
-        # for Windows
+        # for Windows until 6.0.0
         set(ROUTING_SERVICE_API_FOUND TRUE)
     endif()
 

--- a/resources/cmake/FindRTIConnextDDS.cmake
+++ b/resources/cmake/FindRTIConnextDDS.cmake
@@ -1184,18 +1184,20 @@ list(APPEND rti_versions_field_names_target
     "target_libraries"
 )
 
-# Define CONNEXTDDS_INCLUDE_DIRS
-find_path(CONNEXTDDS_INCLUDE_DIRS
+find_path(CONNEXTDDS_BASE_INCLUDE_DIRS
     NAMES
         "ndds_c.h"
     PATHS
         "${CONNEXTDDS_DIR}/include/ndds"
 )
 
-set(CONNEXTDDS_INCLUDE_DIRS
+# Define CONNEXTDDS_INCLUDE_DIRS
+set(CONNEXTDDS_INCLUDE_DIRS)
+
+list(APPEND CONNEXTDDS_INCLUDE_DIRS
     "${CONNEXTDDS_DIR}/include"
-    ${CONNEXTDDS_INCLUDE_DIRS}
-    "${CONNEXTDDS_INCLUDE_DIRS}/hpp"
+    ${CONNEXTDDS_BASE_INCLUDE_DIRS}
+    "${CONNEXTDDS_BASE_INCLUDE_DIRS}/hpp"
 )
 
 # Find all flavors of nddscore
@@ -1474,9 +1476,9 @@ if(security_plugins IN_LIST RTIConnextDDS_FIND_COMPONENTS)
 
         # Add OpenSSL include directories to the list of
         # CONNEXTDDS_INCLUDE_DIRS
-        set(CONNEXTDDS_INCLUDE_DIRS
+        list(APPEND CONNEXTDDS_INCLUDE_DIRS
             "${OPENSSL_INCLUDE_DIR}"
-            ${CONNEXTDDS_INCLUDE_DIRS})
+        )
 
         # Add OpenSSL libraries to the list of CONNEXTDDS_EXTERNAL_LIBS
         set(CONNEXTDDS_EXTERNAL_LIBS

